samples: boards: pm on stm32 boards with serial wakeup

FRASTM opened the pull request:
This commit retrains the power management testcase
tests/samples/boards/stm32/power_mgmt/serial_wakeup
to bun on stm32 boards only.
Limiting first to nucleo_wg55 and stm32l562dk targets,
more to be added.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/49268

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
samples: boards: pm on stm32 boards with serial wakeup

This commit retrains the power management testcase
tests/samples/boards/stm32/power_mgmt/serial_wakeup
to bun on stm32 boards only.
Limiting first to nucleo_wg55 and stm32l562dk targets,
more to be added.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

hunterhu3000 commented:
The patch looks good on mec15. The test case is skipped as expected
```
DEBUG   - platform filter: ['mec15xxevb_assy6853']
DEBUG   -     arch_filter: None
DEBUG   -      tag_filter: None
DEBUG   -     exclude_tag: None
INFO    - Building initial testsuite list...
DEBUG   - mec15xxevb_assy6853       samples/boards/stm32/power_mgmt/serial_wakeup/sample.boards.stm32.power_mgmt.serial_wakeup SKIPPED: Not in testsuite platform allow list
INFO    - Writing JSON report /home/ztest/zephyrproject/zephyr/twister-out/testplan.json
```

gmarull commented on samples/boards/stm32/power_mgmt/serial_wakeup/sample.yaml:
```diff
@@ -15,3 +15,4 @@ tests:
       - nucleo_wb55rg
     filter: dt_compat_enabled("zephyr,power-state")
     extra_args: "CONFIG_DEBUG=y"
+    platform_allow: nucleo_wb55rg stm32l562e_dk
```
why don't you filter using stm32 uart compatible instead?

erwango replied:
Agreed.
This test will only work on the boards that are provided with an overlay to set up a clock configuration compatible with stop modes.

erwango commented:
Please also remove check on `        - "Device ready"` in yaml file.
"Device ready" is no more printed since https://github.com/zephyrproject-rtos/zephyr/pull/49167

FRASTM committed:
samples: boards: stm32 serial_wakeup does not check device ready

Since the change in the PR  #49167
samples: boards: stm32: serial_wakeup: fix device usage issues
the sample should not check the "Device ready" anymore

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
ADd a commit to "remove check on - "Device ready" in yaml file."
